src/main/java/com/google/jenkins/plugins/computeengine/InstanceConfiguration.java

```diff
@@ -236,6 +245,19 @@ public void setCreateSnapshot(boolean createSnapshot) {
         this.createSnapshot = createSnapshot && this.oneShot;
     }
 
+    /**
+     * Required for JCasC Compatibility,
```

Comment:
JCasC never guarantees backwards compatibility, and users should be aware of that. IMO fix the test instead.

Reply:
Ok, currently removed this setter method. And accordingly fixed the casc yaml files used in tests. With that the tests are now working.

So, we need to inform the users in the release notes of this plugin to take action as,

if preemptible was set
change: `preemptible: true` to `provisioningType: PreemptibleVm`


if preemptible was not set,
simply remove `preemptible: false`

Reply:
I heard that the plugin the release notes are generated from git commit messages,

So when this PR is being merged, maybe I should ensure that it is `Squash` merged, with ensuring the above guidelines for fixing the JCasC yaml files is mentioned in the merge message. 

Right?

or is there a different process to be followed for such breaking change communications?

Reply:
At a minimum, the plugin maintainer should apply the `breaking` label.

If we _can_ offer JCasC compatibility without unreasonable effort, IMO we should. Otherwise it is an extra step for users to take, and in cases like this where the replacement is a newly introduced syntax, it may be difficult to arrange logistically if their plugin versions are not also pinned in the same SCM.

Reply:
ok, noted about the label usages.

currently JCasC compatibility is also kept, so no issues for this PR.